docs/README.md

```diff
@@ -0,0 +1,51 @@
+# PostHog Documentation
+
+Developer-focused documentation alongside code. Update docs in the same PR as your code changes.
+
+## Structure
+
+### `published/` - Published on posthog.com
+
+Documentation published to https://posthog.com/handbook/engineering/ when merged to master:
+
+- Architecture guides
+- Contributing guides
+- Engineering practices
+- Runbooks for self-hosters
+
+**When to use**: Documentation that external users, contributors, or self-hosters need.
+
+### `internal/` - GitHub-only
+
+Documentation that stays in the repository:
+
+- Development workflows
+- Migration patterns
+- Team processes
+
+**When to use**: Knowledge useful for the team but not for external users. We're open source, so "internal" means GitHub-only rather than truly private.
```

Comment:
Do we want to talk about the runbook publicly at all? I can see how a new engineer might only read this and get confused about runbooks which is private